tidy3d/components/data/dataset.py

```diff
@@ -354,13 +355,21 @@ class ModeSolverDataset(ElectromagneticFieldDataset):
         description="Complex-valued effective propagation constants associated with the mode.",
     )
 
-    n_group_raw: ModeIndexDataArray = pd.Field(
+    n_group_raw: GroupIndexDataArray = pd.Field(
         None,
         alias="n_group",
         title="Group Index",
         description="Index associated with group velocity of the mode.",
     )
 
+    dispersion_raw: ModeDispersionDataArray = pd.Field(
+        None,
+        alias="dispersion",
```

Comment:
The alias was introduced because there was an original `n_group` field, which we replaced by `n_group_raw` so that the user would get a warning when accessing a missing `n_group` attribute. By then, the alias was required to load old json/hdf5 data with the `n_group` tag.

I kept the same logic for dispersion: issuing a warning when accessing the possibly empty attribute and storing the actual data in a `_raw` field. I guess the alias is not really necessary here, that's an oversight.